ffac-update-location-gps/files/etc/hotplug.d/usb/10-update-location-gps

```diff
@@ -0,0 +1,26 @@
+#!/bin/sh
+TTYPATH="/sys${DEVPATH}/tty"
+LOCKPATH="/tmp/update-location-gps"
+
+ENABLED=$(uci get update-location-gps.settings.enabled)
+
+if [ "${ACTION}" = "bind" ]; then
+	[ "${ENABLED}" != "1" ] && exit 0
+	test -e "${LOCKPATH}" && exit 0
+	if test -e  "${TTYPATH}"; then
+		TTY_NAME=$(find "${TTYPATH}" -mindepth 1 -maxdepth 1 -type d 2>/dev/null | sed -n 's|.*/tty/||p')
```

Comment:
On my machine, this seems to yield several TTY devices and not a single device. Is this expected? Or did I use the wrong path?
```
# find /sys/devices/virtual/tty -mindepth 1 -maxdepth 1 -type d | sed -n 's|.*/tty/||p' | wc -l
515
```

A `ls ${TTYPATH}*` would probably be easier and doesn't require `sed`.

As this is a local variable, I would also make it lowercase instead, so shellcheck can check its usage across the file.

Reply:
The  usb path in DEVPATH is something like  `/devices/platform/1f400000.fpi/1e101000.usb/usb1/1-1/1-1:1.0`  so there exists `/sys/devices/platform/1f400000.fpi/1e101000.usb/usb1/1-1/1-1:1.0/tty/ttyACM0`.
From this, one can use `cat /dev/ttyACM0` to access the output of the USB GPS device.

Though there exists the TTY section for hotplugs, which seems more fitting here:
https://openwrt.org/docs/guide-user/base-system/hotplug#tty

So I will switch to this

Reply:
Now we have a lock file per TTY which waits.
This results in one process per TTY which read the other terminals to see if they also have GPS output - as I do not have a better way to detect if the TTY has this capability.

I don't think that this is a very good way to deploy this package for all models, but it seems the least harmful way which supports all cases of GPS devices.

Reply:
> I don't think that this is a very good way to deploy this package for all models

How would you change it?